### DISCUSSION TITLE
How to configure golangci-lint to detect out-of-bound slice access?

### DISCUSSION THREAD

**Discussion by cacosub7** [golangci/golangci-lint#4150] [Q&A]
Hi all,
I'm using golangci-lint(*version v1.54.1 built with go1.21.0*) and need to detect out-of-bound slice access in my Go code. Example:
```
type Item struct {
    Name  string
    Value int
}

sourceItems := []*Item{
    {"item1", 10},
    {"item2", 20},
    {"item3", 30},
}

destinationItems := make([]*Item, 0, len(sourceItems))

i := 0
for _, item := range sourceItems {
    destinationItems[i] = item // Expect out-of-range linter warning here
    i++
}
```
I've tried `enable-all: true` and `gosec` 's *G602* rule , but they don't catch this case.

Any suggestions for a linter or golangci-lint configuration that detects out-of-bound access and warns on the problematic line?

Thanks!

**Comment by hendrywiranto** [comment]
Hi, trying to help answer here
I think the kind of error you're having here should not be the linter responsibility to catch since linter is a static program analysis after all

also your example error should be caught on the compile time, but I think you're suggesting that the out-of-range is when we are allocating the size on the runtime dynamically (?)
